### DISCUSSION TITLE
Cannot join voice channel correctly

### DISCUSSION THREAD

**Discussion by ryanbester** [bwmarrin/discordgo#1338] [Q&A]
I am trying to connect to a voice channel using `ChannelVoiceJoin`. This has been working correctly but when I executed the command for the bot to join the voice channel the bot hangs. I had a look and the `ChannelVoiceJoin` doesn't appear to ever return.

I increased the `LogLevel` to Debug and observed that there was no opcode 4 (Session Description) being received from `onEvent` in voice.go. Oddly, when I changed the voice region, to somewhere such as Brazil, opcode 4 was correctly being received and the `ChannelVoiceJoin` returned correctly.

I'm not sure if this is a problem with the library or just my internet connection. I don't have another internet connection to test, but it's strange how certain voice regions work correctly.

This is my code:
```go
vc, err := s.ChannelVoiceJoin(i.GuildID, vcId, false, false)
if err != nil {
_ = SendErrorResponseEdit(s, i, "Error joining", "Failed to connect to voice channel.")
return
}

channel, err := s.Channel(vcId)
if err != nil {
_ = SendErrorResponseEdit(s, i, "Error joining", "Invalid voice channel.")
return
}

_, _ = s.InteractionResponseEdit(i.Interaction, &discordgo.WebhookEdit{
Embeds: &[]*discordgo.MessageEmbed{
	{
		Title:       "Joined voice channel",
		Description: fmt.Sprintf("Joined `%s`", channel.Name),
		Color:       core.AppContext.Config.EmbedColor,
	},
},
})
```

Go: 1.18.6, discordgo: 0.27.0
I have tried on Windows and Linux and experience the same behaviour.

Please could anyone advise me as to why this may be the case? And if it is maybe just my internet connection.

**Comment by imkunet** [comment]
This seems to be a known issue, and is a breaking change introduced by Discord. See https://github.com/DV8FromTheWorld/JDA/commit/4bb15efefeebd92f333d7a8d30cb6b99f229ef32 for a potential solution ?

**Reply by FedorLap2006** [reply]
This should be fixed in `v0.27.1`. However, looking at the dates, the change should have not been rolled out.

**Comment by Psycos66** [comment]
I had the same problem, but after searching a very long time, I have found out that, when i was using `go run`, it was not working... Then I have tried to build my project with `go build`, and it worked.

It was Avast, who blocked the connection. For me turning it off and using go build helped.
